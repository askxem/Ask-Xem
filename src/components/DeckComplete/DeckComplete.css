.backdrop {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.7);
  z-index: 4;
  display: flex;
  justify-content: center;
  align-items: center;
}

.buttonContainer {
  display: flex;
  justify-content: flex-end;
}

.exitButton {
  width: 15%;
  cursor: pointer;
}

.completedContent {
  border-radius: 10px;
  z-index: 4;
  width: 70vw;
  padding: 5%;
  padding-bottom: 10%;
  text-overflow: scroll;
  background-color: white;
  font-family: 'Varela Round', sans-serif;
  color: black;
  justify-content: center;
}

.buttonDiv {
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-around;
  gap: 5%;
}

.link {
  width: 60%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  border: none;
  background-color: #91ad8c;
  border-radius: 20px;
  padding: 10px;
  box-shadow: 0px 5px 10px gray;
  color: white;
}

.link:hover {
  color: #fffa9b;
}

@media only screen and (min-width: 768px) {
  .link {
    width: 40%;
  }

  .exitButton {
    width: 5%;
  }
}
