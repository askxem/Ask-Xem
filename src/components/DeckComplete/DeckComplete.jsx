import { motion, AnimatePresence } from 'framer-motion'
import Confetti from 'react-confetti'
import styles from './DeckComplete.css'
import { Link } from 'react-router-dom'

// initial state of the backdrop
const backdropVariants = {
  visible: { opacity: 1 },
  hidden: { opacity: 0 }
}

export default function DeckComplete({ handleClick }) {


  return (
    <AnimatePresence>
      <motion.div
        className={styles.backdrop}
        variants={backdropVariants}
        initial='hidden'
        animate='visible'
      >
        <motion.div className={styles.completeModal}>
          <Confetti />
          <div className={styles.completedContent}>
            <div className={styles.buttonContainer}>
              <svg onClick={handleClick} xmlns="http://www.w3.org/2000/svg" className={styles.exitButton} fill="none" viewBox="0 0 24 24" stroke="currentColor">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" />
              </svg>
            </div>

            <p>You completed the rainbow!</p>
            <p>Great job!</p>
            <p>Now would be a great time to open the journal to keep track of what you learned or how you feel. Or you can take a look at the favorites you've been saving along the way!</p>

            <div className={styles.buttonDiv}>
              <Link to={`/favorites`} className={styles.link}>Favorites</Link>
              <Link to={`/select`} className={styles.link}>Choose New Deck</Link>
            </div>
          </div>
        </motion.div>
      </motion.div>

    </AnimatePresence>
  )
}
