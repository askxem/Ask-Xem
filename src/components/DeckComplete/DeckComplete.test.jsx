import {render} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import DeckComplete from './DeckComplete.jsx';


it.only('renders the deck complete component', () => {
    const {container} = render(
      <MemoryRouter>
            <DeckComplete />
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot();
})