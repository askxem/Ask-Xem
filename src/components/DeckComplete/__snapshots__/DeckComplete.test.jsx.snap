// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the deck complete component 1`] = `
<div>
  <div
    class="backdrop"
    style="opacity: 0;"
  >
    <div
      class="completeModal"
    >
      <canvas
        height="768"
        style="z-index: 2; position: absolute; pointer-events: none; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        width="1024"
      />
      <div
        class="completedContent"
      >
        <div
          class="buttonContainer"
        >
          <svg
            class="exitButton"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p>
          You completed the rainbow!
        </p>
        <p>
          Great job!
        </p>
        <p>
          Now would be a great time to open the journal to keep track of what you learned or how you feel. Or you can take a look at the favorites you've been saving along the way!
        </p>
        <div
          class="buttonDiv"
        >
          <a
            class="link"
            href="/favorites"
          >
            Favorites
          </a>
          <a
            class="link"
            href="/select"
          >
            Choose New Deck
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
